### PR TITLE
Fix: missing UUID while generating deps

### PR DESCRIPTION
**Pull request by vdayanand**

### CODE REVIEW

File: src/utils/pkgtools.jl [hunk 1]
```diff
@@ -177,7 +143,11 @@ function dependencies_per_package(registry=joinpath(homedir(), ".julia/registrie
                     if VersionNumber(version) in Pkg.Types.VersionRange(compatver)
                         for (dep, vers) in compattoml[compatver]
                             depdict = get!(deps, dep) do
-                                uuid = string(get(stdlib_to_uuid, dep, ""))
+                                uuid  = if haskey(name_uuid_map, dep)
+                                    string(name_uuid_map[dep])
+                                else
+                                    error("UUID not found for $dep")
```
Comment (pfitzseb) [inline src/utils/pkgtools.jl:149]: Let's only `@error` here and break out of the loop, I think.

File: src/utils/pkgtools.jl [hunk 2]
```diff
@@ -73,46 +73,12 @@ is_jll(name) = endswith(name, "_jll")
...
-        )
-
+    name_to_uuid = Dict()
     for (uuid, pkg) in allpkgs
-        uuid_to_name[UUID(uuid)] = pkg.name
+        name_to_uuid[UUID(uuid)] = pkg.name
     end
-
-    uuid_to_name
+    merge!(name_to_uuid, stdlib_to_uuid)
```
Comment (pfitzseb) [inline src/utils/pkgtools.jl:80]: This isn't right. One maps uuids to package names and the other one maps package names to uuids.

File: src/utils/pkgtools.jl (hunk 1 above)
Comment (vdayanand) [reply src/utils/pkgtools.jl:149]: oops. fixed it

File: src/utils/pkgtools.jl (hunk 2 above)
Comment (vdayanand) [reply src/utils/pkgtools.jl:80]: oops. fixed it

File: src/utils/pkgtools.jl (hunk 1 above)
Comment (pfitzseb) [reply src/utils/pkgtools.jl:149]: Actually, I don't think we need the `haskey` check at all. We can give up straight away if there is no `pkgname` => uuid mapping in Deps.toml even though `pkgname` is referenced in Compat.toml.

Comment (vdayanand) [reply src/utils/pkgtools.jl]: I am not sure I get this point. I think Deps.toml must have `pkgname` => `uuid` mapping however same is not true for Compat.toml as packages there are referred by just their package name

Comment (pfitzseb) [reply src/utils/pkgtools.jl]: Right. The issue here is (I think) that you can have entries in Compat.toml that aren't mentioned in Deps.toml at all. That doesn't really break anything and we should just ignore it.

**Pull request merged**